src/components/PeopleFilters.tsx

```diff
@@ -1,18 +1,56 @@
+import { useSearchParams } from 'react-router-dom';
+import { SearchLink } from './SearchLink';
+import classNames from 'classnames';
+import { getSearchWith } from '../utils/searchHelper';
+
 export const PeopleFilters = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const sex = searchParams.get('sex') || '';
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  function setSearchWidth(params: any) {
+    const search = getSearchWith(searchParams, params);
+
+    setSearchParams(search);
+  }
+
+  const handleQueryChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchWidth({ query: e.target.value || null });
+  };
+
+  const toggleCenturies = (ct: string) => {
+    const newCentr = centuries.includes(ct)
+      ? centuries.filter(century => century !== ct)
+      : [...centuries, ct];
+
+    return { centuries: newCentr };
```

Comment:
The toggleCenturies function returns an object with the updated centuries, but it is not directly updating the search parameters. You might want to call setSearchWidth with the returned object to update the search parameters accordingly.